--- a/include/sta_info.h	2026-03-15 20:12:35.881778892 +0700
+++ b/include/sta_info.h	2026-03-15 20:12:33.068078838 +0700
@@ -511,8 +511,13 @@
 
 	struct st_ctl_t st_ctl;
 	u8 max_agg_num_minimal_record; /*keep minimal tx desc max_agg_num setting*/
+  
 	u8 curr_rx_rate;
 	u8 curr_rx_rate_bmc;
+  
+  u8 last_rx_rate; /* simpan rate terakhir RX STA ini */
+  u8 last_tx_rate; /* simpan rate terakhir TX STA ini */
+
 #ifdef CONFIG_RTS_FULL_BW
 	bool vendor_8812;
 #endif
