--- a/hal/rtl8188e/rtl8188e_rxdesc.c	2025-12-11 08:56:43.000000000 +0700
+++ b/hal/rtl8188e/rtl8188e_rxdesc.c	2026-03-15 22:40:38.750178960 +0700
@@ -64,7 +64,11 @@
 		pattrib->mfrag = (u8)((report.rxdw1 >> 27) & 0x1);/* (u8)prxreport->mf; */
 		pattrib->mdata = (u8)((report.rxdw1 >> 26) & 0x1);/* (u8)prxreport->md; */
 
-		pattrib->data_rate = (u8)(report.rxdw3 & 0x3f);/* (u8)prxreport->rxmcs; */
+    /* Ambil bit 0-6 untuk nilai rate, dan bit 7 sebagai penanda HT (MCS) */
+    pattrib->data_rate = (u8)(report.rxdw3 & 0x7f);/* (u8)prxreport->rxmcs; */
+    if (report.rxdw3 & BIT(7)) {
+        pattrib->data_rate |= 0x80; /* Tandai sebagai HT rate */
+    }
 
 		pattrib->icv_err = (u8)((report.rxdw0 >> 15) & 0x1);/* (u8)prxreport->icverr; */
 		pattrib->shift_sz = (u8)((report.rxdw0 >> 24) & 0x3);
